// @across-protocol/contracts/contract-types/ethers/factories/ArbitrumCrossDomainEnabled__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  ArbitrumCrossDomainEnabled,
  ArbitrumCrossDomainEnabledInterface,
} from "../ArbitrumCrossDomainEnabled";

const _abi = [
  {
    inputs: [],
    name: "inbox",
    outputs: [
      {
        internalType: "contract iArbitrum_Inbox",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class ArbitrumCrossDomainEnabled__factory {
  static readonly abi = _abi;
  static createInterface(): ArbitrumCrossDomainEnabledInterface {
    return new utils.Interface(_abi) as ArbitrumCrossDomainEnabledInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ArbitrumCrossDomainEnabled {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as ArbitrumCrossDomainEnabled;
  }
}
