// @across-protocol/contracts/contract-types/ethers/factories/AVMCrossDomainEnabled__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  AVMCrossDomainEnabled,
  AVMCrossDomainEnabledInterface,
} from "../AVMCrossDomainEnabled";

const _abi = [
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        internalType: "address",
        name: "from",
        type: "address",
      },
      {
        indexed: true,
        internalType: "address",
        name: "to",
        type: "address",
      },
      {
        indexed: true,
        internalType: "uint256",
        name: "id",
        type: "uint256",
      },
      {
        indexed: false,
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
    ],
    name: "SentCrossDomainMessage",
    type: "event",
  },
];

export class AVMCrossDomainEnabled__factory {
  static readonly abi = _abi;
  static createInterface(): AVMCrossDomainEnabledInterface {
    return new utils.Interface(_abi) as AVMCrossDomainEnabledInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): AVMCrossDomainEnabled {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as AVMCrossDomainEnabled;
  }
}
