// @across-protocol/contracts/contract-types/ethers/factories/BridgePoolInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  BridgePoolInterface,
  BridgePoolInterfaceInterface,
} from "../BridgePoolInterface";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "newAdmin",
        type: "address",
      },
    ],
    name: "changeAdmin",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [],
    name: "l1Token",
    outputs: [
      {
        internalType: "contract IERC20",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint64",
        name: "_newLpFeeRatePerSecond",
        type: "uint64",
      },
    ],
    name: "setLpFeeRatePerSecond",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bool",
        name: "_relaysEnabled",
        type: "bool",
      },
    ],
    name: "setRelaysEnabled",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class BridgePoolInterface__factory {
  static readonly abi = _abi;
  static createInterface(): BridgePoolInterfaceInterface {
    return new utils.Interface(_abi) as BridgePoolInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): BridgePoolInterface {
    return new Contract(address, _abi, signerOrProvider) as BridgePoolInterface;
  }
}
