// @across-protocol/contracts/contract-types/ethers/factories/IGenericHandler__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  IGenericHandler,
  IGenericHandlerInterface,
} from "../IGenericHandler";

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "resourceID",
        type: "bytes32",
      },
      {
        internalType: "address",
        name: "contractAddress",
        type: "address",
      },
      {
        internalType: "bytes4",
        name: "depositFunctionSig",
        type: "bytes4",
      },
      {
        internalType: "bytes4",
        name: "executeFunctionSig",
        type: "bytes4",
      },
    ],
    name: "setResource",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class IGenericHandler__factory {
  static readonly abi = _abi;
  static createInterface(): IGenericHandlerInterface {
    return new utils.Interface(_abi) as IGenericHandlerInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IGenericHandler {
    return new Contract(address, _abi, signerOrProvider) as IGenericHandler;
  }
}
