// @across-protocol/contracts/contract-types/ethers/factories/MessengerInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  MessengerInterface,
  MessengerInterfaceInterface,
} from "../MessengerInterface";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "target",
        type: "address",
      },
      {
        internalType: "address",
        name: "userToRefund",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "l1CallValue",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "gasLimit",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "gasPrice",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "maxSubmissionCost",
        type: "uint256",
      },
      {
        internalType: "bytes",
        name: "message",
        type: "bytes",
      },
    ],
    name: "relayMessage",
    outputs: [],
    stateMutability: "payable",
    type: "function",
  },
];

export class MessengerInterface__factory {
  static readonly abi = _abi;
  static createInterface(): MessengerInterfaceInterface {
    return new utils.Interface(_abi) as MessengerInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): MessengerInterface {
    return new Contract(address, _abi, signerOrProvider) as MessengerInterface;
  }
}
