// @across-protocol/contracts/contract-types/ethers/factories/ParentMessengerConsumerInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  ParentMessengerConsumerInterface,
  ParentMessengerConsumerInterfaceInterface,
} from "../ParentMessengerConsumerInterface";

const _abi = [
  {
    inputs: [
      {
        internalType: "uint256",
        name: "chainId",
        type: "uint256",
      },
      {
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
    ],
    name: "processMessageFromChild",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class ParentMessengerConsumerInterface__factory {
  static readonly abi = _abi;
  static createInterface(): ParentMessengerConsumerInterfaceInterface {
    return new utils.Interface(
      _abi
    ) as ParentMessengerConsumerInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ParentMessengerConsumerInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as ParentMessengerConsumerInterface;
  }
}
