// @across-protocol/contracts/contract-types/ethers/factories/ParentMessengerInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  ParentMessengerInterface,
  ParentMessengerInterfaceInterface,
} from "../ParentMessengerInterface";

const _abi = [
  {
    inputs: [],
    name: "getL1CallValue",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
    ],
    name: "sendMessageToChild",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class ParentMessengerInterface__factory {
  static readonly abi = _abi;
  static createInterface(): ParentMessengerInterfaceInterface {
    return new utils.Interface(_abi) as ParentMessengerInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ParentMessengerInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as ParentMessengerInterface;
  }
}
