// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/Balancer__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Balancer__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "tokenIn",
                type: "address",
            },
            {
                internalType: "address",
                name: "tokenOut",
                type: "address",
            },
        ],
        name: "getSpotPriceSansFee",
        outputs: [
            {
                internalType: "uint256",
                name: "spotPrice",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class Balancer__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.Balancer__factory = Balancer__factory;
Balancer__factory.abi = _abi;
