// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/MinimumVotingAncillaryInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MinimumVotingAncillaryInterface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "voterAddress",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "roundId",
                type: "uint256",
            },
            {
                components: [
                    {
                        internalType: "bytes32",
                        name: "identifier",
                        type: "bytes32",
                    },
                    {
                        internalType: "uint256",
                        name: "time",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "ancillaryData",
                        type: "bytes",
                    },
                ],
                internalType: "struct MinimumVotingAncillaryInterface.PendingRequestAncillary[]",
                name: "toRetrieve",
                type: "tuple[]",
            },
        ],
        name: "retrieveRewards",
        outputs: [
            {
                components: [
                    {
                        internalType: "uint256",
                        name: "rawValue",
                        type: "uint256",
                    },
                ],
                internalType: "struct MinimumVotingAncillaryInterface.Unsigned",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class MinimumVotingAncillaryInterface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.MinimumVotingAncillaryInterface__factory = MinimumVotingAncillaryInterface__factory;
MinimumVotingAncillaryInterface__factory.abi = _abi;
