// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/ParentMessengerConsumerInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParentMessengerConsumerInterface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "uint256",
                name: "chainId",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "data",
                type: "bytes",
            },
        ],
        name: "processMessageFromChild",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ParentMessengerConsumerInterface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ParentMessengerConsumerInterface__factory = ParentMessengerConsumerInterface__factory;
ParentMessengerConsumerInterface__factory.abi = _abi;
