// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/SkinnyOptimisticOracleV2Interface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkinnyOptimisticOracleV2Interface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "ancillaryBytesLimit",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
        ],
        name: "disputePrice",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "disputer",
                type: "address",
            },
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
        ],
        name: "disputePriceFor",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
        ],
        name: "getState",
        outputs: [
            {
                internalType: "enum OptimisticOracleInterface.State",
                name: "",
                type: "uint8",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
        ],
        name: "hasPrice",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
            {
                internalType: "int256",
                name: "proposedPrice",
                type: "int256",
            },
        ],
        name: "proposePrice",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "proposer",
                type: "address",
            },
            {
                internalType: "int256",
                name: "proposedPrice",
                type: "int256",
            },
        ],
        name: "proposePriceFor",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                internalType: "contract IERC20",
                name: "currency",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "reward",
                type: "uint256",
            },
            {
                components: [
                    {
                        internalType: "bool",
                        name: "callbackOnPriceProposed",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "callbackOnPriceDisputed",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "callbackOnPriceSettled",
                        type: "bool",
                    },
                    {
                        internalType: "uint256",
                        name: "bond",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "customLiveness",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                name: "requestSettings",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "proposer",
                type: "address",
            },
            {
                internalType: "int256",
                name: "proposedPrice",
                type: "int256",
            },
        ],
        name: "requestAndProposePriceFor",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                internalType: "contract IERC20",
                name: "currency",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "reward",
                type: "uint256",
            },
            {
                components: [
                    {
                        internalType: "bool",
                        name: "callbackOnPriceProposed",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "callbackOnPriceDisputed",
                        type: "bool",
                    },
                    {
                        internalType: "bool",
                        name: "callbackOnPriceSettled",
                        type: "bool",
                    },
                    {
                        internalType: "uint256",
                        name: "bond",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "customLiveness",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                name: "requestSettings",
                type: "tuple",
            },
        ],
        name: "requestPrice",
        outputs: [
            {
                internalType: "uint256",
                name: "totalBond",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "identifier",
                type: "bytes32",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                components: [
                    {
                        internalType: "address",
                        name: "proposer",
                        type: "address",
                    },
                    {
                        internalType: "address",
                        name: "disputer",
                        type: "address",
                    },
                    {
                        internalType: "contract IERC20",
                        name: "currency",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "settled",
                        type: "bool",
                    },
                    {
                        components: [
                            {
                                internalType: "bool",
                                name: "callbackOnPriceProposed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceDisputed",
                                type: "bool",
                            },
                            {
                                internalType: "bool",
                                name: "callbackOnPriceSettled",
                                type: "bool",
                            },
                            {
                                internalType: "uint256",
                                name: "bond",
                                type: "uint256",
                            },
                            {
                                internalType: "uint256",
                                name: "customLiveness",
                                type: "uint256",
                            },
                        ],
                        internalType: "struct SkinnyOptimisticOracleV2Interface.RequestSettings",
                        name: "requestSettings",
                        type: "tuple",
                    },
                    {
                        internalType: "int256",
                        name: "proposedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "int256",
                        name: "resolvedPrice",
                        type: "int256",
                    },
                    {
                        internalType: "uint256",
                        name: "expirationTime",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "reward",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "finalFee",
                        type: "uint256",
                    },
                ],
                internalType: "struct SkinnyOptimisticOracleV2Interface.Request",
                name: "request",
                type: "tuple",
            },
        ],
        name: "settle",
        outputs: [
            {
                internalType: "uint256",
                name: "payout",
                type: "uint256",
            },
            {
                internalType: "int256",
                name: "resolvedPrice",
                type: "int256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes",
                name: "ancillaryData",
                type: "bytes",
            },
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
        ],
        name: "stampAncillaryData",
        outputs: [
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
];
class SkinnyOptimisticOracleV2Interface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.SkinnyOptimisticOracleV2Interface__factory = SkinnyOptimisticOracleV2Interface__factory;
SkinnyOptimisticOracleV2Interface__factory.abi = _abi;
