// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/TransactionBatcher__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionBatcher__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address[]",
                name: "targets",
                type: "address[]",
            },
            {
                internalType: "uint256[]",
                name: "values",
                type: "uint256[]",
            },
            {
                internalType: "bytes[]",
                name: "datas",
                type: "bytes[]",
            },
        ],
        name: "batchSend",
        outputs: [],
        stateMutability: "payable",
        type: "function",
    },
];
class TransactionBatcher__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.TransactionBatcher__factory = TransactionBatcher__factory;
TransactionBatcher__factory.abi = _abi;
