// @uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/factories/VotingInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VotingInterface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        name: "getVoterFromDelegate",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        name: "voterStakes",
        outputs: [
            {
                components: [
                    {
                        internalType: "uint128",
                        name: "stake",
                        type: "uint128",
                    },
                    {
                        internalType: "uint128",
                        name: "pendingUnstake",
                        type: "uint128",
                    },
                    {
                        internalType: "uint128",
                        name: "rewardsPaidPerToken",
                        type: "uint128",
                    },
                    {
                        internalType: "uint128",
                        name: "outstandingRewards",
                        type: "uint128",
                    },
                    {
                        internalType: "int128",
                        name: "unappliedSlash",
                        type: "int128",
                    },
                    {
                        internalType: "uint64",
                        name: "nextIndexToProcess",
                        type: "uint64",
                    },
                    {
                        internalType: "uint64",
                        name: "unstakeTime",
                        type: "uint64",
                    },
                    {
                        internalType: "address",
                        name: "delegate",
                        type: "address",
                    },
                ],
                internalType: "struct VotingInterface.VoterStake",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class VotingInterface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.VotingInterface__factory = VotingInterface__factory;
VotingInterface__factory.abi = _abi;
