// @uma/contracts-node/dist/packages/contracts-node/typechain/@across-protocol/contracts/ethers/factories/FinderInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FinderInterface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "interfaceName",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "implementationAddress",
                type: "address",
            },
        ],
        name: "changeImplementationAddress",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "interfaceName",
                type: "bytes32",
            },
        ],
        name: "getImplementationAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class FinderInterface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.FinderInterface__factory = FinderInterface__factory;
FinderInterface__factory.abi = _abi;
