// @uma/contracts-node/dist/packages/contracts-node/typechain/@across-protocol/contracts/ethers/factories/IGenericHandler__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IGenericHandler__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "resourceID",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "contractAddress",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "depositFunctionSig",
                type: "bytes4",
            },
            {
                internalType: "bytes4",
                name: "executeFunctionSig",
                type: "bytes4",
            },
        ],
        name: "setResource",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IGenericHandler__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IGenericHandler__factory = IGenericHandler__factory;
IGenericHandler__factory.abi = _abi;
