// @gnosis.pm/zodiac/dist/abi-typechain-types/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Usul__factory = exports.Tellor__factory = exports.ScopeGuard__factory = exports.Roles__factory = exports.RealityEth__factory = exports.RealityErc20__factory = exports.Permissions__factory = exports.OzGovernor__factory = exports.OptimisticGovernor__factory = exports.MultisendEncoder__factory = exports.ModuleProxyFactory__factory = exports.MetaGuard__factory = exports.ExitErc721__factory = exports.ExitErc20__factory = exports.Erc721Votes__factory = exports.Erc20Votes__factory = exports.Delay__factory = exports.Connext__factory = exports.CirculatingSupplyErc721__factory = exports.CirculatingSupplyErc20__factory = exports.Bridge__factory = exports.factories = void 0;
exports.factories = __importStar(require("./factories"));
var Bridge__factory_1 = require("./factories/Bridge__factory");
Object.defineProperty(exports, "Bridge__factory", { enumerable: true, get: function () { return Bridge__factory_1.Bridge__factory; } });
var CirculatingSupplyErc20__factory_1 = require("./factories/CirculatingSupplyErc20__factory");
Object.defineProperty(exports, "CirculatingSupplyErc20__factory", { enumerable: true, get: function () { return CirculatingSupplyErc20__factory_1.CirculatingSupplyErc20__factory; } });
var CirculatingSupplyErc721__factory_1 = require("./factories/CirculatingSupplyErc721__factory");
Object.defineProperty(exports, "CirculatingSupplyErc721__factory", { enumerable: true, get: function () { return CirculatingSupplyErc721__factory_1.CirculatingSupplyErc721__factory; } });
var Connext__factory_1 = require("./factories/Connext__factory");
Object.defineProperty(exports, "Connext__factory", { enumerable: true, get: function () { return Connext__factory_1.Connext__factory; } });
var Delay__factory_1 = require("./factories/Delay__factory");
Object.defineProperty(exports, "Delay__factory", { enumerable: true, get: function () { return Delay__factory_1.Delay__factory; } });
var Erc20Votes__factory_1 = require("./factories/Erc20Votes__factory");
Object.defineProperty(exports, "Erc20Votes__factory", { enumerable: true, get: function () { return Erc20Votes__factory_1.Erc20Votes__factory; } });
var Erc721Votes__factory_1 = require("./factories/Erc721Votes__factory");
Object.defineProperty(exports, "Erc721Votes__factory", { enumerable: true, get: function () { return Erc721Votes__factory_1.Erc721Votes__factory; } });
var ExitErc20__factory_1 = require("./factories/ExitErc20__factory");
Object.defineProperty(exports, "ExitErc20__factory", { enumerable: true, get: function () { return ExitErc20__factory_1.ExitErc20__factory; } });
var ExitErc721__factory_1 = require("./factories/ExitErc721__factory");
Object.defineProperty(exports, "ExitErc721__factory", { enumerable: true, get: function () { return ExitErc721__factory_1.ExitErc721__factory; } });
var MetaGuard__factory_1 = require("./factories/MetaGuard__factory");
Object.defineProperty(exports, "MetaGuard__factory", { enumerable: true, get: function () { return MetaGuard__factory_1.MetaGuard__factory; } });
var ModuleProxyFactory__factory_1 = require("./factories/ModuleProxyFactory__factory");
Object.defineProperty(exports, "ModuleProxyFactory__factory", { enumerable: true, get: function () { return ModuleProxyFactory__factory_1.ModuleProxyFactory__factory; } });
var MultisendEncoder__factory_1 = require("./factories/MultisendEncoder__factory");
Object.defineProperty(exports, "MultisendEncoder__factory", { enumerable: true, get: function () { return MultisendEncoder__factory_1.MultisendEncoder__factory; } });
var OptimisticGovernor__factory_1 = require("./factories/OptimisticGovernor__factory");
Object.defineProperty(exports, "OptimisticGovernor__factory", { enumerable: true, get: function () { return OptimisticGovernor__factory_1.OptimisticGovernor__factory; } });
var OzGovernor__factory_1 = require("./factories/OzGovernor__factory");
Object.defineProperty(exports, "OzGovernor__factory", { enumerable: true, get: function () { return OzGovernor__factory_1.OzGovernor__factory; } });
var Permissions__factory_1 = require("./factories/Permissions__factory");
Object.defineProperty(exports, "Permissions__factory", { enumerable: true, get: function () { return Permissions__factory_1.Permissions__factory; } });
var RealityErc20__factory_1 = require("./factories/RealityErc20__factory");
Object.defineProperty(exports, "RealityErc20__factory", { enumerable: true, get: function () { return RealityErc20__factory_1.RealityErc20__factory; } });
var RealityEth__factory_1 = require("./factories/RealityEth__factory");
Object.defineProperty(exports, "RealityEth__factory", { enumerable: true, get: function () { return RealityEth__factory_1.RealityEth__factory; } });
var Roles__factory_1 = require("./factories/Roles__factory");
Object.defineProperty(exports, "Roles__factory", { enumerable: true, get: function () { return Roles__factory_1.Roles__factory; } });
var ScopeGuard__factory_1 = require("./factories/ScopeGuard__factory");
Object.defineProperty(exports, "ScopeGuard__factory", { enumerable: true, get: function () { return ScopeGuard__factory_1.ScopeGuard__factory; } });
var Tellor__factory_1 = require("./factories/Tellor__factory");
Object.defineProperty(exports, "Tellor__factory", { enumerable: true, get: function () { return Tellor__factory_1.Tellor__factory; } });
var Usul__factory_1 = require("./factories/Usul__factory");
Object.defineProperty(exports, "Usul__factory", { enumerable: true, get: function () { return Usul__factory_1.Usul__factory; } });
