// @uma/core/dist/contract-types/ethers/factories/DSNote__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DSNote__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: true,
        inputs: [
            {
                indexed: true,
                internalType: "bytes4",
                name: "sig",
                type: "bytes4",
            },
            {
                indexed: true,
                internalType: "address",
                name: "guy",
                type: "address",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "foo",
                type: "bytes32",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "bar",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes",
                name: "fax",
                type: "bytes",
            },
        ],
        name: "LogNote",
        type: "event",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea264697066735822122095edc756f8e54ee26f347d6392240d1fd5f89c0ee6c877d49d1e6f07059a16fa64736f6c63430008100033";
class DSNote__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (args.length === 1) {
            super(_abi, _bytecode, args[0]);
        }
        else {
            super(...args);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.DSNote__factory = DSNote__factory;
DSNote__factory.bytecode = _bytecode;
DSNote__factory.abi = _abi;
