// @uma/core/dist/contract-types/ethers/factories/OptimisticOracleV3CallbackRecipientInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptimisticOracleV3CallbackRecipientInterface__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "assertionId",
                type: "bytes32",
            },
        ],
        name: "assertionDisputedCallback",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "assertionId",
                type: "bytes32",
            },
            {
                internalType: "bool",
                name: "assertedTruthfully",
                type: "bool",
            },
        ],
        name: "assertionResolvedCallback",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class OptimisticOracleV3CallbackRecipientInterface__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.OptimisticOracleV3CallbackRecipientInterface__factory = OptimisticOracleV3CallbackRecipientInterface__factory;
OptimisticOracleV3CallbackRecipientInterface__factory.abi = _abi;
