// @uma/core/dist/contract-types/ethers/factories/Stakeable__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stakeable__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                indexed: true,
                internalType: "address",
                name: "newMember",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "manager",
                type: "address",
            },
        ],
        name: "AddedSharedMember",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                indexed: true,
                internalType: "address",
                name: "oldMember",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "manager",
                type: "address",
            },
        ],
        name: "RemovedSharedMember",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                indexed: true,
                internalType: "address",
                name: "newMember",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "manager",
                type: "address",
            },
        ],
        name: "ResetExclusiveMember",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "newMember",
                type: "address",
            },
        ],
        name: "addMember",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "votingContract",
                type: "address",
            },
        ],
        name: "executeUnstake",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
        ],
        name: "getMember",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "memberToCheck",
                type: "address",
            },
        ],
        name: "holdsRole",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "memberToRemove",
                type: "address",
            },
        ],
        name: "removeMember",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
        ],
        name: "renounceMembership",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint128",
                name: "amount",
                type: "uint128",
            },
            {
                internalType: "address",
                name: "votingContract",
                type: "address",
            },
        ],
        name: "requestUnstake",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "roleId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "newMember",
                type: "address",
            },
        ],
        name: "resetMember",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint128",
                name: "amount",
                type: "uint128",
            },
            {
                internalType: "address",
                name: "votingContract",
                type: "address",
            },
        ],
        name: "stake",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
        ],
        name: "withdraw",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "erc20Address",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
        ],
        name: "withdrawErc20",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class Stakeable__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.Stakeable__factory = Stakeable__factory;
Stakeable__factory.abi = _abi;
