// @across-protocol/contracts/contract-types/web3/FxBaseRootTunnel.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export interface FxBaseRootTunnel extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): FxBaseRootTunnel;
  clone(): FxBaseRootTunnel;
  methods: {
    SEND_MESSAGE_EVENT_SIG(): NonPayableTransactionObject<string>;

    checkpointManager(): NonPayableTransactionObject<string>;

    fxChildTunnel(): NonPayableTransactionObject<string>;

    fxRoot(): NonPayableTransactionObject<string>;

    processedExits(
      arg0: string | number[]
    ): NonPayableTransactionObject<boolean>;

    receiveMessage(
      inputData: string | number[]
    ): NonPayableTransactionObject<void>;

    setFxChildTunnel(_fxChildTunnel: string): NonPayableTransactionObject<void>;
  };
  events: {
    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };
}
