// @across-protocol/contracts/contract-types/web3/MessengerInterface.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export interface MessengerInterface extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): MessengerInterface;
  clone(): MessengerInterface;
  methods: {
    relayMessage(
      target: string,
      userToRefund: string,
      l1CallValue: number | string | BN,
      gasLimit: number | string | BN,
      gasPrice: number | string | BN,
      maxSubmissionCost: number | string | BN,
      message: string | number[]
    ): PayableTransactionObject<void>;
  };
  events: {
    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };
}
