// @across-protocol/contracts/contract-types/web3/Optimism_Wrapper.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type ChangedBridgePool = ContractEventLog<{
  bridgePool: string;
  0: string;
}>;
export type OwnershipTransferred = ContractEventLog<{
  previousOwner: string;
  newOwner: string;
  0: string;
  1: string;
}>;

export interface OptimismWrapper extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): OptimismWrapper;
  clone(): OptimismWrapper;
  methods: {
    bridgePool(): NonPayableTransactionObject<string>;

    changeBridgePool(newBridgePool: string): NonPayableTransactionObject<void>;

    owner(): NonPayableTransactionObject<string>;

    renounceOwnership(): NonPayableTransactionObject<void>;

    transferOwnership(newOwner: string): NonPayableTransactionObject<void>;

    weth(): NonPayableTransactionObject<string>;

    wrapAndTransfer(): PayableTransactionObject<void>;
  };
  events: {
    ChangedBridgePool(cb?: Callback<ChangedBridgePool>): EventEmitter;
    ChangedBridgePool(
      options?: EventOptions,
      cb?: Callback<ChangedBridgePool>
    ): EventEmitter;

    OwnershipTransferred(cb?: Callback<OwnershipTransferred>): EventEmitter;
    OwnershipTransferred(
      options?: EventOptions,
      cb?: Callback<OwnershipTransferred>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "ChangedBridgePool", cb: Callback<ChangedBridgePool>): void;
  once(
    event: "ChangedBridgePool",
    options: EventOptions,
    cb: Callback<ChangedBridgePool>
  ): void;

  once(event: "OwnershipTransferred", cb: Callback<OwnershipTransferred>): void;
  once(
    event: "OwnershipTransferred",
    options: EventOptions,
    cb: Callback<OwnershipTransferred>
  ): void;
}
