// @across-protocol/contracts/contract-types/web3/ParentMessengerBase.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type OwnershipTransferred = ContractEventLog<{
  previousOwner: string;
  newOwner: string;
  0: string;
  1: string;
}>;
export type SetChildMessenger = ContractEventLog<{
  childMessenger: string;
  0: string;
}>;
export type SetGovernorHub = ContractEventLog<{
  governorHub: string;
  0: string;
}>;
export type SetGovernorSpoke = ContractEventLog<{
  governorSpoke: string;
  0: string;
}>;
export type SetOracleHub = ContractEventLog<{
  oracleHub: string;
  0: string;
}>;
export type SetOracleSpoke = ContractEventLog<{
  oracleSpoke: string;
  0: string;
}>;

export interface ParentMessengerBase extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): ParentMessengerBase;
  clone(): ParentMessengerBase;
  methods: {
    childChainId(): NonPayableTransactionObject<string>;

    childMessenger(): NonPayableTransactionObject<string>;

    getL1CallValue(): NonPayableTransactionObject<string>;

    governorHub(): NonPayableTransactionObject<string>;

    governorSpoke(): NonPayableTransactionObject<string>;

    oracleHub(): NonPayableTransactionObject<string>;

    oracleSpoke(): NonPayableTransactionObject<string>;

    owner(): NonPayableTransactionObject<string>;

    renounceOwnership(): NonPayableTransactionObject<void>;

    sendMessageToChild(
      data: string | number[]
    ): NonPayableTransactionObject<void>;

    setChildMessenger(
      newChildMessenger: string
    ): NonPayableTransactionObject<void>;

    setGovernorHub(newGovernorHub: string): NonPayableTransactionObject<void>;

    setGovernorSpoke(
      newGovernorSpoke: string
    ): NonPayableTransactionObject<void>;

    setOracleHub(newOracleHub: string): NonPayableTransactionObject<void>;

    setOracleSpoke(newOracleSpoke: string): NonPayableTransactionObject<void>;

    transferOwnership(newOwner: string): NonPayableTransactionObject<void>;
  };
  events: {
    OwnershipTransferred(cb?: Callback<OwnershipTransferred>): EventEmitter;
    OwnershipTransferred(
      options?: EventOptions,
      cb?: Callback<OwnershipTransferred>
    ): EventEmitter;

    SetChildMessenger(cb?: Callback<SetChildMessenger>): EventEmitter;
    SetChildMessenger(
      options?: EventOptions,
      cb?: Callback<SetChildMessenger>
    ): EventEmitter;

    SetGovernorHub(cb?: Callback<SetGovernorHub>): EventEmitter;
    SetGovernorHub(
      options?: EventOptions,
      cb?: Callback<SetGovernorHub>
    ): EventEmitter;

    SetGovernorSpoke(cb?: Callback<SetGovernorSpoke>): EventEmitter;
    SetGovernorSpoke(
      options?: EventOptions,
      cb?: Callback<SetGovernorSpoke>
    ): EventEmitter;

    SetOracleHub(cb?: Callback<SetOracleHub>): EventEmitter;
    SetOracleHub(
      options?: EventOptions,
      cb?: Callback<SetOracleHub>
    ): EventEmitter;

    SetOracleSpoke(cb?: Callback<SetOracleSpoke>): EventEmitter;
    SetOracleSpoke(
      options?: EventOptions,
      cb?: Callback<SetOracleSpoke>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "OwnershipTransferred", cb: Callback<OwnershipTransferred>): void;
  once(
    event: "OwnershipTransferred",
    options: EventOptions,
    cb: Callback<OwnershipTransferred>
  ): void;

  once(event: "SetChildMessenger", cb: Callback<SetChildMessenger>): void;
  once(
    event: "SetChildMessenger",
    options: EventOptions,
    cb: Callback<SetChildMessenger>
  ): void;

  once(event: "SetGovernorHub", cb: Callback<SetGovernorHub>): void;
  once(
    event: "SetGovernorHub",
    options: EventOptions,
    cb: Callback<SetGovernorHub>
  ): void;

  once(event: "SetGovernorSpoke", cb: Callback<SetGovernorSpoke>): void;
  once(
    event: "SetGovernorSpoke",
    options: EventOptions,
    cb: Callback<SetGovernorSpoke>
  ): void;

  once(event: "SetOracleHub", cb: Callback<SetOracleHub>): void;
  once(
    event: "SetOracleHub",
    options: EventOptions,
    cb: Callback<SetOracleHub>
  ): void;

  once(event: "SetOracleSpoke", cb: Callback<SetOracleSpoke>): void;
  once(
    event: "SetOracleSpoke",
    options: EventOptions,
    cb: Callback<SetOracleSpoke>
  ): void;
}
