// @across-protocol/contracts/contract-types/web3/RateModelStore.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type OwnershipTransferred = ContractEventLog<{
  previousOwner: string;
  newOwner: string;
  0: string;
  1: string;
}>;
export type UpdatedRateModel = ContractEventLog<{
  l1Token: string;
  rateModel: string;
  0: string;
  1: string;
}>;

export interface RateModelStore extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): RateModelStore;
  clone(): RateModelStore;
  methods: {
    l1TokenRateModels(arg0: string): NonPayableTransactionObject<string>;

    multicall(data: (string | number[])[]): PayableTransactionObject<string[]>;

    owner(): NonPayableTransactionObject<string>;

    renounceOwnership(): NonPayableTransactionObject<void>;

    transferOwnership(newOwner: string): NonPayableTransactionObject<void>;

    updateRateModel(
      l1Token: string,
      rateModel: string
    ): NonPayableTransactionObject<void>;
  };
  events: {
    OwnershipTransferred(cb?: Callback<OwnershipTransferred>): EventEmitter;
    OwnershipTransferred(
      options?: EventOptions,
      cb?: Callback<OwnershipTransferred>
    ): EventEmitter;

    UpdatedRateModel(cb?: Callback<UpdatedRateModel>): EventEmitter;
    UpdatedRateModel(
      options?: EventOptions,
      cb?: Callback<UpdatedRateModel>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "OwnershipTransferred", cb: Callback<OwnershipTransferred>): void;
  once(
    event: "OwnershipTransferred",
    options: EventOptions,
    cb: Callback<OwnershipTransferred>
  ): void;

  once(event: "UpdatedRateModel", cb: Callback<UpdatedRateModel>): void;
  once(
    event: "UpdatedRateModel",
    options: EventOptions,
    cb: Callback<UpdatedRateModel>
  ): void;
}
