// @across-protocol/contracts/contract-types/web3/StateSyncMock.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import BN from "bn.js";
import { ContractOptions } from "web3-eth-contract";
import { EventLog } from "web3-core";
import { EventEmitter } from "events";
import {
  Callback,
  PayableTransactionObject,
  NonPayableTransactionObject,
  BlockType,
  ContractEventLog,
  BaseContract,
} from "./types";

export interface EventOptions {
  filter?: object;
  fromBlock?: BlockType;
  topics?: string[];
}

export type StateSynced = ContractEventLog<{
  id: string;
  contractAddress: string;
  data: string;
  0: string;
  1: string;
  2: string;
}>;

export interface StateSyncMock extends BaseContract {
  constructor(
    jsonInterface: any[],
    address?: string,
    options?: ContractOptions
  ): StateSyncMock;
  clone(): StateSyncMock;
  methods: {
    syncState(
      receiver: string,
      data: string | number[]
    ): NonPayableTransactionObject<void>;
  };
  events: {
    StateSynced(cb?: Callback<StateSynced>): EventEmitter;
    StateSynced(
      options?: EventOptions,
      cb?: Callback<StateSynced>
    ): EventEmitter;

    allEvents(options?: EventOptions, cb?: Callback<EventLog>): EventEmitter;
  };

  once(event: "StateSynced", cb: Callback<StateSynced>): void;
  once(
    event: "StateSynced",
    options: EventOptions,
    cb: Callback<StateSynced>
  ): void;
}
