// @gnosis.pm/zodiac/dist/abi-typechain-types/factories/MultisendEncoder__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultisendEncoder__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "NoTransactions",
        type: "error",
    },
    {
        inputs: [],
        name: "UnequalArraysLengths",
        type: "error",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "multisend",
                type: "address",
            },
            {
                internalType: "address[]",
                name: "targets",
                type: "address[]",
            },
            {
                internalType: "uint256[]",
                name: "values",
                type: "uint256[]",
            },
            {
                internalType: "bytes[]",
                name: "calldatas",
                type: "bytes[]",
            },
        ],
        name: "encodeMultisend",
        outputs: [
            {
                internalType: "address",
                name: "to",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "value",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "data",
                type: "bytes",
            },
            {
                internalType: "enum Enum.Operation",
                name: "operation",
                type: "Enum.Operation",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
];
class MultisendEncoder__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.MultisendEncoder__factory = MultisendEncoder__factory;
MultisendEncoder__factory.abi = _abi;
